src/test/java/org/kohsuke/github/GHRepositoryForkBuilderTest.java

```diff
@@ -0,0 +1,235 @@
+    private void verifyBranches(GHRepository forked, boolean defaultBranchOnly) throws IOException {
+        Map<String, GHBranch> branches = forked.getBranches();
+        if (defaultBranchOnly) {
+            assertThat(branches.size(), equalTo(1));
+        } else {
+            assertThat(branches.size(), greaterThan(1));
+        }
+        assertThat(branches.containsKey(forked.getDefaultBranch()), is(true));
+    }
+
+    /**
+     * Test fork.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void testFork() throws Exception {
+        // GHRepository forkedRepo = repo.createFork().create();
+        GHRepository forkedRepo = repo.fork();
+
+        verifyBasicForkProperties(repo, forkedRepo, repo.getName());
+        verifyBranches(forkedRepo, false);
+
+        forkedRepo.delete();
+    }
+
+    /**
+     * Test fork to org.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void testForkToOrg() throws Exception {
+        GHOrganization targetOrg = gitHub.getOrganization(TARGET_ORG);
+        // GHRepository forkedRepo = repo.createFork().organization(targetOrg).create();
+        GHRepository forkedRepo = repo.forkTo(targetOrg);
+
+        verifyBasicForkProperties(repo, forkedRepo, repo.getName());
+        verifyBranches(forkedRepo, false);
+
+        forkedRepo.delete();
+    }
+
+    /**
+     * Test fork default branch only.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void testForkDefaultBranchOnly() throws Exception {
+        GHRepository forkedRepo = repo.createFork().defaultBranchOnly(true).create();
+
+        verifyBasicForkProperties(repo, forkedRepo, repo.getName());
+        verifyBranches(forkedRepo, true);
+
+        forkedRepo.delete();
+    }
+
+    /**
+     * Test fork changed name.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void testForkChangedName() throws Exception {
+        String newRepoName = "test-fork-with-new-name";
+        GHRepository forkedRepo = repo.createFork().name(newRepoName).create();
+
+        assertThat(forkedRepo.getName(), equalTo(newRepoName));
+        verifyBasicForkProperties(repo, forkedRepo, newRepoName);
+        verifyBranches(forkedRepo, false);
+
+        forkedRepo.delete();
+    }
+
+    /**
+     * Test timeout message and sleep count.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void testTimeoutMessage() throws Exception {
+        // repo.createFork().name("test-message").create();
+        class TrackingSleepBuilder extends GHRepositoryForkBuilder {
+            int sleepCount = 0;
+            int lastSleepMillis = 0;
+
+            TrackingSleepBuilder(GHRepository repo) {
+                super(repo);
+            }
+
+            @Override
+            protected void sleep(int millis) throws IOException {
```

Comment:
This should be package scoped, not protected.  

`protected` makes the method available for override externally.  You don't want that. You only want the tests to be able to override it. 

```suggestion
            void sleep(int millis) throws IOException {
```